src/ui/map.js

```diff
@@ -1077,31 +1076,21 @@ class Map extends Camera {
     }
 
     _updateProjection(explicitProjection?: ProjectionSpecification | null): this {
-        const prevProjection = this.getProjection();
         if (explicitProjection === null) { this._explicitProjection = null; }
         const projection = explicitProjection || this.getProjection();
 
-        // At high zoom on globe, set transform projection to Mercator while _explicitProjection stays globe.
-        const newProjection = this.transform.setProjection(projection && projection.name === 'globe' ?
-            {name: (this.transform.zoom >= GLOBE_ZOOM_THRESHOLD_MAX ? 'mercator' : 'globe')} :
-            projection);
+        const newProjection = this.transform.setProjection(projection);
```

Comment:
Nit: A bit of extra redundancy to set globe to Mercator even if it's added back to `ProjectionSpecification`.

```suggestion
        const newProjection = this.transform.setProjection(projection && projection.name === 'globe' ?  'mercator' : projection);
```

Reply:
I think `projection.name === 'globe'` might not ever make it to this point since it'll already be rejected as an error, so in my opinion, this probably isn't necessary. Can you explain more about adding this check and what you are thinking? Thank ya for the review!

Reply:
It won't, but this provides a another catch even if a user undoes the changes to `projection/index.js`, to create behavior consistent with https://github.com/mapbox/mapbox-gl-js/pull/11405.

As I mentioned in our call, changes to this file aren't really necessary at all because of the error. But the change you've made here doesn't disable globe, merely the transition to Mercator.

Reply:
Just tested this out. My previous suggestion isn't quite right, it needs to be:
```suggestion
        const newProjection = this.transform.setProjection(projection && projection.name === 'globe' ?  {name: 'mercator'} : projection);
```
It works as I describe above, but it does trigger a flow error. IMHO this isn't worth spending more time on, I'm happy to merge as is.

Reply:
ah ok thanks @SnailBones !